sqlmesh/core/model/meta.py

```diff
@@ -391,3 +395,16 @@ def when_matched(self) -> t.Optional[exp.When]:
         if isinstance(self.kind, IncrementalByUniqueKeyKind):
             return self.kind.when_matched
         return None
+
+    @property
+    def catalog(self) -> t.Optional[str]:
+        """Returns the catalog of a model."""
+        return self.fully_qualified_table.catalog
+
+    @property
+    def fully_qualified_table(self) -> exp.Table:
+        return exp.to_table(self.fqn)
```

Comment:
i know this is moving out of meta, but make sure it's cached so to_table isn't called a million types